### PR TITLE
fix broken m2r dependency, fix wrong API description, added Example

### PR DESCRIPTION
**Pull request by MalteIwanicki**
* The dependency [m2r](https://github.com/miyakogi/m2r) is not maintained since a long time ago and now leads to a conflict. I changed the m2r dependency to it's still maintained [m2r2 ](https://github.com/CrossNox/m2r2) fork.

* The description for the API endpoint `PATCH api/vs/datasets/{id}` was written under `PUT`. I fixed the location and added an example. (I couldn't get the part underneath the moved one to work so that might be also wrong. But I didn't have time to investigate that)

* I added examples for how to get metadata and how to update them.

### CODE REVIEW

**Comment by giohappy** [comment]
Thanks for the contribution @MalteIwanicki!

What about moving the example right before [Resource Upload](https://docs.geonode.org/en/master/devel/api/integration/index.html#resource-upload)? 
Also, the endpoint more correctly retrieves the "Resource Details". An example of how to retrieve the actual standard metadata could be added, which involves a request to the CSW service with the right parameters and the resource UUID.

E.g.

- Dublin Core: https://master.demo.geonode.org/catalogue/csw?request=GetRecordById&service=CSW&version=2.0.2&id=454dff93-354e-4221-a659-e69998ff1e97&outputschema=http://www.isotc211.org/2005/gmd&elementsetname=full
- ISO: https://master.demo.geonode.org/catalogue/csw?request=GetRecordById&service=CSW&version=2.0.2&id=454dff93-354e-4221-a659-e69998ff1e97&outputschema=http://www.isotc211.org/2005/gmd&elementsetname=full

File: devel/api/integration/index.rst [hunk 1]
```diff
@@ -533,3 +533,54 @@ The ``status_url`` property returns the URL to track kthe progress of the reques
 
 The operation will be completed once the ``status`` property is updated with the value ``finished``.
 
+Dataset Get Metadata
```
Comment (gannebamm) [inline devel/api/integration/index.rst:536]: rename to 'Dataset Get standardized Metadata'

and follow @giohappy s comment https://github.com/GeoNode/documentation/pull/264#issuecomment-1316817796

> An example of how to retrieve the actual standard metadata could be added, which can be obtained from the `links` array inside the detail response.

Retrieving the Ressource Details is already described here (Example 2.)
https://docs.geonode.org/en/master/devel/api/integration/index.html#example-requests

**Comment by gannebamm** [comment]
> What about moving the example right before [Resource Upload](https://docs.geonode.org/en/master/devel/api/integration/index.html#resource-upload)?

The Get Metadata part should be moved up there (as described in my comment above) but the Update examples are better suited here: https://docs.geonode.org/en/master/devel/api/integration/index.html#resource-update

I think @giohappy ?

**Comment by giohappy** [comment]
>I changed the title and name. My Idea was to not mix the api/v2/resources/ and api/v2/datasets/ but I think both is fine.
@MalteIwanicki Actually the details can be obtained also from the resources/ endpoint. They return more or less the same information (except for things specific to datasets).
Maybe it would be good to clarify this.

**Comment by ahmdthr** [comment]
It would also be a good idea to mention that bbox_polygon should not be explicitly PATCHed, because it is a computed property and will be overwritten again on the next update. In fact, it should be discouraged to PATCH bbox_polygon.

**Comment by MalteIwanicki** [comment]
@giohappy Note added.
@ahmdthr thanks for pointing that out

**Comment by MalteIwanicki** [comment]
@giohappy I opened this [issue](https://github.com/GeoNode/geonode/issues/10316) for that. And created a [PR](https://github.com/GeoNode/geonode/pull/10317).
This all together should be good for a merge hopefully.

**Comment by MalteIwanicki** [comment]
@afabiani, @giohappy, @mattiagiupponi 
The depending PR was merged therefore this can be merged as well.

**Pull request merged**